### ISSUE TITLE
Incorrect Topic viewing target

### ISSUE THREAD

**Issue by fosskers** [magit/forge#368]
If:

1. You have Magit open in a project that Forge knows about.
2. You have pulled that project's topics.
3. You open the special Forge Notification buffer and attempt to visit a topic from a _different_ repo.
4. The topic number of that issue happens to coincide with one from the repo you're in.

Then Forge will navigate you to the topic of that number in the repo you're in, not the one you asked for.

For example, I had Forge's magit open, opened the special Notification buffer, attempted to visit `#36 WASM Support` of the `EmbarkStudios/tame-oauth` repo, and was brought to issue #36 of Forge instead.

Note that this only occurs from visting a Topic from the special Forge Notification buffer, _not_ from the `Issues` dropdown in the normal Magit Status buffer.

**Comment by tarsius** [comment]
Yeah, 58ce1627ba308ea3d62fb63948aac515a40d9c0e and 64f9a5bb4a10a1603191913ad1f1b4ecc0811844 were a mistake. I'll have to think about how to best address that. Maybe there's an alternative to just reverting this.

**Comment by tarsius** [comment]
Tadaa!
